test.py

```diff
@@ -775,19 +777,27 @@ def __init__(self, test_no: int, shortname: str, suite) -> None:
         self.server_log_filename: Optional[pathlib.Path] = None
         PythonTest._reset(self)
 
+    def _prepare_pytest_params(self, options: argparse.Namespace):
+        self.args = [
+            "-s",  # don't capture print() output inside pytest
+            "--log-level=DEBUG",   # Capture logs
+            "-o",
+            "junit_family=xunit2",
+            "--junit-xml={}".format(self.xmlout)]
+        if options.markers:
+            self.args.append(f"-m={options.markers}")
+
+            # https://docs.pytest.org/en/7.1.x/reference/exit-codes.html
+            no_tests_selected_exit_code = 5
+            self.valid_exit_codes = [0, no_tests_selected_exit_code]
```

Comment:
Some subdirectories of `scylla/test` may not contain `@slow` tests, so running `scylla/test.py --markers slow` will fail since a single run of `test.py` executes `pytest` for all subdirectories. A better way to handle this would be to check that among all runs of `pytest` at least one returned `exit code != 5`, but this is somewhat new behavior that must be handled for all types of tests, including C++ ones, and it is not obvious if it is worth doing, so I left this workaround for now. This problem hasn't arisen before since there was no way to filter running tests, and each directory contains at least one test.